samcli/commands/_utils/template.py

```diff
@@ -29,20 +30,26 @@ class TemplateFailedParsingException(UserException):
     pass
 
 
-def get_template_data(template_file):
+def get_template_data(template_file, root_template_dir: Optional[str] = None):
     """
     Read the template file, parse it as JSON/YAML and return the template as a dictionary.
 
     Parameters
     ----------
     template_file : string
         Path to the template to read
+    root_template_dir: string
+        Optional directory of the root SAM Template
 
     Returns
     -------
     Template data as a dictionary
     """
 
+    # if template_file is a relative path , we use root_template_dir to convert to absolute path
+    if not os.path.isabs(template_file) and root_template_dir:
+        template_file = os.path.join(root_template_dir, template_file)
```

Comment:
Is there any way to do this without requiring multiple variables to specify a single path?

e.g. in one location make the path absolute, and pass that around, so that callees just read from the path without caring whether it's absolute or relative

Reply:
I think that is possible and useful. It needs a relative larger refactoring since currently we have a few places doing path checking and converting. I'd say it's out of scope for this bug and we can do this in a separate PR.

Reply:
Any way to reduce the number of places where these path modifications are done?

It's already somewhat confusing as for example `normalize_resource_path` already does some path changes, but then further changes are done here.

Reply:
Discussed elsewhere: `normalize_resource_path` no longer converts to absolute path only since a bug was discovered. Agreed to add comment explaining the context.